src/averageAges.js

```diff
@@ -38,6 +49,16 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const woman = withChildren
+    ? people
+      .filter((person, i, array) => person.sex === 'f'
+        && array.some(a => a.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+  const womanAvgAge = woman
+    .reduce((memo, person) => memo + (person.died - person.born), 0)
+      / woman.length;
```

Comment:
You use this iteration a few times, so you can make a separate function with this action.